sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -2458,6 +2516,192 @@ checkDevSupportDeviceRequirements(const device &Dev,
     }
   }
 
+  // TODO: remove checks for CUDA and HIP from if-statement below when runtime
```

Comment:
Overall, this is a lot of new lines - should this be outlined into a separate helper to avoid having endless functions?

Reply:
Moved most part of the code to `checkDevSupportJointMatrix` and `checkDevSupportJointMatrixMad` functions